core/src/main/java/org/infinispan/stats/impl/ContainerStatsFactory.java

```diff
@@ -0,0 +1,15 @@
+package org.infinispan.stats.impl;
+
+import org.infinispan.factories.AbstractComponentFactory;
+import org.infinispan.factories.AutoInstantiableFactory;
+import org.infinispan.factories.annotations.DefaultFactoryFor;
+import org.infinispan.stats.ContainerStats;
+
+@DefaultFactoryFor(classes = ContainerStats.class)
+public class ContainerStatsFactory extends AbstractComponentFactory implements AutoInstantiableFactory {
```

Comment:
@jabolina I'm not getting NPE in some tests as `TestingUtil.extractComponent(cache, ClusterCacheStats.class)` is returning null. It's been a long time since I worked with the factories, have I missed a step?

Reply:
I think reverting the `ClusterCacheStatsFactory` should fix everything. The *CacheStats has a cache scope, so it still needs its own factory. The *ContainerStats have a global scope, we can move them to the `EmptyConstructorFactory`.

Reply:
O man, it seems I can't read anymore. I was confusing `ClusterCacheStats` with `ClusterContainerStats` when I was debugging :man_facepalming:. Should be fixed now, thanks!